var/spack/repos/builtin/packages/henson/package.py

```diff
@@ -14,10 +14,13 @@ class Henson(CMakePackage):
 
     version("master", branch="master")
 
+    maintainers("mrzv")
+
     depends_on("mpi")
 
     variant("python", default=False, description="Build Python bindings")
     extends("python", when="+python")
+    depends_on("py-mpi4py", when="+python")
```

Comment:
What dependency type should this be? I assume it's needed at build/run-time, not sure about link-time.

Reply:
That's an interesting question. As far as I know it's only build and run-time, there is nothing that's being linked from mpi4py.

Reply:
You can check the build log and see. But if it's build/run only, you can use:
```suggestion
    depends_on("py-mpi4py", when="+python", type=("build", "run"))
```

Reply:
What does that buy? I.e., why does it ever make sense to have build without link?

Reply:
(In any case, I've added this change.)

Reply:
These types have the following meaning:

* build: add to PATH/PYTHONPATH during `spack install`
* run: add to PATH/PYTHONPATH during `spack load` or `module load` or `spack env activate`
* link: add to link/include commands and inject RPATHs via compiler wrappers
* test: only build when `spack install --test`

Reply:
Ah, makes perfect sense. Thanks for the explanation.